java/algorithm/src/test/java/org/apache/arrow/algorithm/sort/TestFixedWidthSorting.java

```diff
@@ -31,64 +31,47 @@
 import org.apache.arrow.vector.IntVector;
 import org.apache.arrow.vector.SmallIntVector;
 import org.apache.arrow.vector.TinyIntVector;
-import org.junit.After;
-import org.junit.Before;
-import org.junit.Test;
-import org.junit.runner.RunWith;
-import org.junit.runners.Parameterized;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
 
 /**
  * Test sorting fixed width vectors with random data.
  */
-@RunWith(Parameterized.class)
 public class TestFixedWidthSorting<V extends BaseFixedWidthVector, U extends Comparable<U>> {
 
   static final int[] VECTOR_LENGTHS = new int[] {2, 5, 10, 50, 100, 1000, 3000};
 
   static final double[] NULL_FRACTIONS = {0, 0.1, 0.3, 0.5, 0.7, 0.9, 1};
 
-  private final int length;
-
-  private final double nullFraction;
-
-  private final boolean inPlace;
-
-  private final Function<BufferAllocator, V> vectorGenerator;
-
-  private final TestSortingUtil.DataGenerator<V, U> dataGenerator;
-
   private BufferAllocator allocator;
 
-  @Before
+  @BeforeEach
   public void prepare() {
     allocator = new RootAllocator(Integer.MAX_VALUE);
   }
 
-  @After
+  @AfterEach
   public void shutdown() {
     allocator.close();
   }
 
-  public TestFixedWidthSorting(
-      int length, double nullFraction, boolean inPlace, String desc,
+  @ParameterizedTest(name = "in place = {0}, length = {1}, null fraction = {2}, desc = {3}, vector = {4}")
+  @MethodSource("getParameters")
+  public void testSort(
+      boolean inPlace, int length, double nullFraction, String desc,
```

Comment:
Is the `desc` really needed?

Also, `TestVariableWidthSorting.java` is similar.

Reply:
Seems like it is not being used. IDE should give a hint?

Reply:
I saw a warning message like this: `Class 'ComparatorType' is exposed outside its defined visibility scope` from the IDE. We can actually ignore this message.

I rolled back for consistency.

Reply:
Yes, the IDE gives a hint that `Parameter 'desc' is never used`. 

However, I feel that `desc` might add some clarity. I'm not entirely sure whether it should be removed.

```java
          params.add(Arguments.of(
              inPlace, length, nullFrac, "TinyIntVector",
              (Function<BufferAllocator, TinyIntVector>) allocator -> new TinyIntVector("vector", allocator),
              TestSortingUtil.TINY_INT_GENERATOR
          ));

          params.add(Arguments.of(
              inPlace, length, nullFrac, "SmallIntVector",
              (Function<BufferAllocator, SmallIntVector>) allocator -> new SmallIntVector("vector", allocator),
              TestSortingUtil.SMALL_INT_GENERATOR
          ));

          params.add(Arguments.of(
              inPlace, length, nullFrac, "IntVector",
              (Function<BufferAllocator, IntVector>) allocator -> new IntVector("vector", allocator),
              TestSortingUtil.INT_GENERATOR
          ));

          params.add(Arguments.of(
              inPlace, length, nullFrac, "BigIntVector",
              (Function<BufferAllocator, BigIntVector>) allocator -> new BigIntVector("vector", allocator),
              TestSortingUtil.LONG_GENERATOR
          ));

          params.add(Arguments.of(
              inPlace, length, nullFrac, "Float4Vector",
              (Function<BufferAllocator, Float4Vector>) allocator -> new Float4Vector("vector", allocator),
              TestSortingUtil.FLOAT_GENERATOR
          ));

          params.add(Arguments.of(
              inPlace, length, nullFrac, "Float8Vector",
              (Function<BufferAllocator, Float8Vector>) allocator -> new Float8Vector("vector", allocator),
              TestSortingUtil.DOUBLE_GENERATOR
          ));

```

Reply:
I think if it is not used we can remove it for the moment.

cc @lidavidm

Reply:
I would like to keep it.

Reply:
You could make it a named parameter set instead though: https://stackoverflow.com/questions/57892989/generating-display-names-for-parameterizedtest-in-junit-5

Reply:
Yes. We are already at 5.10.2 JUNIT Jupyter dependency. I think that's better.

Reply:
Thank you for your review.

I checked what you said, and it looks like the current code needs to use `argumentSet` to take multiple arguments for each test. This is only available in JUnit `5.11.0` or higher.

So, it seems like it's hard to use Named Parameters right now. Did I understand that correctly?

Additionally, I am working on resolving the conflict. Sorry for the delay; it's because of my main job.

Reply:
@llama90 Please take your time, no rush in our end. 

I thought that feature is available in 5.10.2 merely referring to what is mentioned in the stackoverflow thread.